nixos/modules/hardware/tuxedo-keyboard.nix

```diff
@@ -0,0 +1,30 @@
+{ config, lib, pkgs, ... }:
+
+# TODO: look at the other hardware modules
+# TODO: re-build for the specific kernel
+# TODO: I am not quite sure IF a module is still needed here since the configurations here could be set directly without this module. 
+# 
+#         boot.kernelParams = [
+#           "tuxedo_keyboard.mode=0"
+#           "tuxedo_keyboard.brightness=${toString cfg.brightness}"
+#           "tuxedo_keyboard.color_left=0x${cfg.color}"
+#         ]; 
```

Comment:
One advantage of having the options as nixos module options is that if they change, it becomes the module's responsibility to keep track of those and not the user.

The minimum would be to document the options, see next comment.

Reply:
I've added some documentations for these options via: 81ccf61d2251d0e84109f3db364f55ccf29bb270